pkg/daemon/update.go

```diff
@@ -1710,6 +1689,34 @@ func (dn *Daemon) updateSSHKeys(newUsers []ign3types.PasswdUser) error {
 func (dn *Daemon) updateOS(config *mcfgv1.MachineConfig, osImageContentDir string) error {
 	newURL := config.Spec.OSImageURL
 	glog.Infof("Updating OS to %s", newURL)
+
+	booted, _, err := dn.NodeUpdaterClient.GetBootedAndStagedDeployment()
+	if err != nil {
+		return err
+	}
+
+	// Before we attempt to do an OS update, we must remove pending RT kernel overrides
+	// because in the case of updating from RHEL8 to RHEL9, the kernel packages are
+	// OS version dependent.  See also https://github.com/coreos/rpm-ostree/issues/2542
+	// (Now really what we want to do here is something more like rpm-ostree override reset --kernel
+	//  i.e. the inverse of https://github.com/coreos/rpm-ostree/pull/4322 so that
+	//  we're again not hardcoding even the prefix of kernel packages)
+	removals := booted.RequestedBaseRemovals
+	kernelOverrides := []string{}
+	for _, removal := range removals {
+		if strings.HasPrefix(removal, "kernel-rt-") {
```

Comment:
Hmm wait I think this needs to be `kernel-`...

Reply:
Ah but wait, if we change that then the fallout is different...it means anyone who has hopped on to a node and is doing `rpm-ostree override replace kernel` and been pinning the kernel will now unconditionally find their overrides removed, when that didn't happen before.

(This "ownership" problem would be fixed with https://github.com/coreos/rpm-ostree/issues/2326 as the MCD could write one file, users could write another; also the MCD would just be translating declarative -> declarative instead of declarative -> imperative)

Hmmm....OK, trying to clean this up more.